src/D2L.CodeStyle.Analyzers/Immutability/ImmutabilityContext.Factory.cs

```diff
@@ -103,12 +104,31 @@ internal partial class ImmutabilityContext {
 					type
 				);
 
-				builder.Add( type, info );
+				extraImmutableTypesBuilder.Add( type, info );
+			}
+
+			// Generate a set of methods that we have specifically determined
+			// have a return value which should be considered Immutable by the Analyzer.
+			var knownImmutableReturnsBuilder = ImmutableHashSet.CreateBuilder<IMethodSymbol>();
+			foreach( ( string typeName, string methodName, string qualifiedAssembly ) in KnownImmutableReturningMethods ) {
+				INamedTypeSymbol type = GetTypeSymbol( compilationAssemblies, compilation, qualifiedAssembly, typeName );
+
+				if( type == null ) {
+					continue;
+				}
+
+				IMethodSymbol methodSymbol = type
+					.GetMembers( methodName )
+					.OfType<IMethodSymbol>()
+					.Single( m => m.Parameters.Length == 0 );
+
+				knownImmutableReturnsBuilder.Add( methodSymbol );

```

Comment:
Hmm, lets be more cautious here, just in case.

```suggestion
				IMethodSymbol[] methodSymbol = type
					.GetMembers( methodName )
					.OfType<IMethodSymbol>()
					.Where( m => m.Parameters.Length == 0 )
					.ToArray();

				knownImmutableReturnsBuilder.Add( methodSymbol[0] );
```